### DISCUSSION TITLE
Option to automatically enable auto-merge for all new PRs

### DISCUSSION THREAD

**Discussion by wrslatz** [community/community#8306] [Pull Requests] [answered]
We currently use the [`palantir/bulldozer` GitHub App](https://github.com/palantir/bulldozer) for automating merges of our pull requests. We'd like to use native GitHub auto-merge, but there doesn't seem to be a way to opt into auto-merge for all new PRs. Users have to manually opt their individual PRs in, increasing the user interactions needed to get PRs merged.

An option to enable auto-merge for all PRs would help. Alternatively, an option is to enable auto-merge programmatically via API call using Actions or a GitHub App.

Also, https://github.com/github/feedback/discussions/6865 is related. Would be nice if all of these features worked together seamlessly and with as few touch-points as possible, once configured.

**Comment by mntlty** [comment, marked as answer]
@wrslatz auto merge can be enabled for a PR using the graphql api https://docs.github.com/en/graphql/reference/mutations#enablepullrequestautomerge

auto-merge requires explicit approval from a user with write access to be enabled, however you could create a GitHub action that will use the above API every time a PR is created / updated.

**Reply by wrslatz** [reply]
Thanks! I think that makes sense. We'll look into automatically enabling for each PR it via an Action or a GitHub App.

**Reply by wrslatz** [reply]
FYI I opened https://github.com/palantir/bulldozer/issues/297 to track supporting this in the Bulldozer GitHub App
